tests/null_object_tests.py

```diff
@@ -87,3 +95,99 @@ class SimplePage(Page):
+        non_exist_el = NonexistentElement()
+
+        first_el_text =non_exist_el.__eq__(page.id_el.text)
+        second_el_text = non_exist_el.__eq__(page.other_el.text)
+        third_el_text = non_exist_el.__eq__(page.another_el.text)
+
+        #print first_el_text == second_el_text
+        self.assertEqual(first_el_text == third_el_text, second_el_text == third_el_text, "NonExistence __eq__ unit test has failed")
+
+    def test_non_existence_nq_element(self):
+        class SimplePage(Page):
+            class_el = Element(Locators.CLASS_NAME, "simple_class")
+            class_xpath_el = Element(Locators.CLASS_NAME, "simple_xpath")
+
+        uri = make_temp_page(ElementTest.page_content)
+        page = SimplePage(self.driver, uri)
+        non_exist_el = NonexistentElement()
+
+        first_el_text = non_exist_el.__ne__(page.class_el.text)
+        second_el_text = non_exist_el.__ne__(page.class_xpath_el.text)
+
+        # print first_el_text == second_el_text
+        self.assertEqual(first_el_text != second_el_text, False,
+                         "NonExistence __ne__ unit test has failed")
+
+        # test when str() is called on the object test that expected parameters are displayed
+
+    def test_non_existence_str_element(self):
+        class SimplePage(Page):
+            class_el = Element(Locators.CLASS_NAME, "simple_class")
+            exception_msg = "Class Name Found"
+            query_string = "key1=value1&key2=value2"
+
+        uri = make_temp_page(ElementTest.page_content)
+        page = SimplePage(self.driver, uri)
+        non_exist_el = NonexistentElement()
+
+        try:
+           result = str(NonexistentElement(page.exception_msg,page.class_el.text,page.query_string))
+           self.assertIsNotNone(result, "NonExistence __str__ unit test has failed")
+        except (WebDriverException) as exception:
+            assert exception.__class__.__name__ == page.exception_msg
+            raise exception
+    # when repr() is called on the object test that expected parameters can be referenced by name. (e.g: object.prop1, object.prop2 etc...)
+
+    def test_non_existence_repr_element(self):
+        class SimplePage(Page):
+            class_el = Element(Locators.CLASS_NAME, "simple_class")
+            exception_msg = "Class Name Found"
+            query_string = "key1=value1&key2=value2"
+
+        uri = make_temp_page(ElementTest.page_content)
+        page = SimplePage(self.driver, uri)
+        non_exist_el = NonexistentElement()
+
+        try:
+            result = repr(NonexistentElement(page.exception_msg, page.class_el.text, page.query_string))
+            self.assertIsNotNone(result,"NonExistence __repr__ unit test has failed")
+        except (WebDriverException) as exception:
+            assert exception.__class__.__name__ == "simple_class"
+            raise exception
+
+    '''test that if an undefined property is referenced, an Exception is thrown that includes the initialization data
```

Comment:
To test this functionality you only need to try accessing an undefined property on the object. e.g:
x = NonexistentElement('val1', 'val2', 'val3')
try:
  x.propThatDoesn'tExist = 0
except Exception as e:
   #Do some assertions against e